url7f43a4752f_shalynnho_pokemonfactory_tgz-pJ8-Networking_ServerJ8/src/Utils/StringUtil.java

```diff
@@ -0,0 +1,628 @@
+	 */
+	public static String trimAfterChars(String str, String chars) {
+		if (str == null) {
+			return null;
+		}
+		if (chars != null) {
+			for (int ix = 0; ix < str.length(); ix++) {
+				for (int jx = 0; jx < chars.length(); jx++) {
+					if (str.charAt(ix) == chars.charAt(jx)) {
+						return str.substring(0, ix);
+					}
+				}
+			}
+		}
+		return str;
+	}
+
+	/** Like indevOf except is case-independent */
+	public static int getIndexIgnoringCase(String str, String subStr) {
+		if (str != null && subStr != null) {
+			return str.toUpperCase().indexOf(subStr.toUpperCase());
+		}
+		return -1;
+	}
+
+	public static boolean endsWithIgnoreCase(String str, String end) {
+		int lend = end.length();
+		return str.regionMatches(true, str.length() - lend, end, 0, lend);
+	}
+
+	/**
+	 * Remove the substring beginning with the final occurrence of the
+	 * separator, if any.
+	 */
+	public static String upToFinal(String str, String sep) {
+		int pos = str.lastIndexOf(sep);
+		if (pos < 0) {
+			return str;
+		}
+		return str.substring(0, pos);
+	}
+
+	/* Return the substring following the final dot */
+	public static String shortName(Object object) {
+		if (object == null) {
+			return null;
+		}
+		String name = object.toString();
+		return name.substring(name.lastIndexOf('.') + 1);
+	}
+
+	/* Return the non-qualified name of the class */
+	public static String shortName(Class clazz) {
+		String className = clazz.getName();
+		return className.substring(className.lastIndexOf('.') + 1);
+	}
+
+	/* Return the non-qualified name of the method (Class.method) */
+	public static String shortName(Method method) {
+		return shortName(method.getDeclaringClass()) + "." + method.getName();
+	}
+
+	/**
+	 * Returns the number of instances of a particular substring in a string.
+	 * This ignores overlap, starting from the left, so 'xxxxxy' would have 2
+	 * instances of 'xx', not 4. Empty string as a substring returns 0.
+	 */
+	public static int countOccurences(String str, String subStr) {
+		int len = subStr.length();
+		if (len == 0) {
+			return 0;
+		}
+		int pos = 0;
+		int count = 0;
+		while ((pos = str.indexOf(subStr, pos)) >= 0) {
+			count++;
+			pos += len;
+		}
+		return count;
+	}
+
+	/* Return a string with all the characters from a reader */
+	public static String fromReader(Reader r) throws IOException {
+		char[] buf = new char[1000];
+		StringBuffer sb = new StringBuffer(1000);
+		int len;
+		while ((len = r.read(buf)) > 0) {
+			sb.append(buf, 0, len);
+		}
+		return sb.toString();
+	}
+
+	/* Return a string with all the characters from an InputStream */
+	public static String fromInputStream(InputStream in) throws IOException {
+		// use our default encoding rather than system default
+		return fromReader(new InputStreamReader(in, Constants.DEFAULT_ENCODING));
+	}
+
+	/** Reads in the entire contents of a file into a string */
+	public static String fromFile(String path) throws IOException {
+		return fromReader(new FileReader(path));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor: *FileReader*. Execution paths that do not contain closure statements were detected. Ensure the object returned by *FileReader()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).